test.go

```diff
@@ -53,3 +54,29 @@ func Test(f FiniteField) bool {
 
 	return true
 }
+
+func FFMTest(f *FFM) bool {
```

Comment:
You might want to write tests checking that an error is thrown when trying to add different dimensions or inverting non-square matrices. 

Also, when a test case fails, you don't want to use fmt.Println but rather t.Failf("your message") 

See how to write proper test files in eg: https://github.com/sachaservan/private-ann/blob/main/pir/db_test.go